Add multiple cdn (#6982)
withastro/docs: src/content/docs/en/reference/configuration-reference.mdx

<p>

> **Type:** `(string | Record.<string, string>)`<br />

Reviewer: ```suggestion
**Type:** `string | Record.<string, string>`<br />
```

Reviewer 2: I'm honestly not sure what's best to use here, but I don't see too many parentheses used, so I'll commit this and we'll figure out the tiny details when this is added to the other PR!  😄

**Default:** `undefined`<br />

> <Since v="2.2.0" />

Reviewer 2: Note to ourselves: this *property* was introduced in v2.2.0 but this *option* will be introduced much later. So we'll have to decide how to let people know, so that someone doesn't try to use this in an earlier version of Astro.

Reviewer 2: Note also to ourselves: I think I have a solution I'm happy with. We can discuss to make sure!

</p>

Specifies the prefix for Astro-generated asset links. This can be used if assets are served from a different domain than the current site.

> <p>
> **If use `string`**
> </p>

Reviewer: ```suggestion
<p>
**If this value is a `string`,**
</p>
```

If you word it like this, you will likely have to reword the following text below it to flow better.

}
```

> <p>
> **If use `Record<string, string>`**
> </p>

Reviewer: ```suggestion
<p>
**If this value is of `Record<string, string>`,**
</p>
```
Same for this

> Applicable to multiple CDN scenarios. If you static file use different CDN, you can configure it like the example below.

Reviewer: Since this is documenting new behaviour, we will need to double check with core for accuracy!

> Applicable to multiple CDN scenarios. If you static file use different CDN, you can configure it like the example below.

Reviewer: ```suggestion
If your static files are served from different CDNs, you can configure Astro to use unique prefix paths for each CDN based on the filetype requested.
```

> For example, if static file is js file, assets prefix use `https://js.cdn.example.com`. If static file is css file, assets prefix use `https://css.cdn.example.com`. Other assets prefix use `https://cdn.example.com`.

Reviewer: ```suggestion
For example, if the static file is a `.js` file, it will be served from `https://js.cdn.example.com`. If it is a `.css` file, it will be served from `https://css.cdn.example.com`. All other assets will be served from `https://cdn.example.com`.
```
Some refactoring

```js
{
  build: {
    assetsPrefix: {
      'js': 'https://js.cdn.example.com',

>       'css': 'https://js.cdn.example.com',

Reviewer: ```suggestion
      'css': 'https://css.cdn.example.com',
```
To match the example in the paragraph

      'defaultAssetsPrefix': 'https://cdn.example.com'
    }
  }
}
```

> Object key is file extension, value is CDN assets prefix path. `defaultAssetsPrefix` is default assets prefix cdn value set. You can set the default values of other files by setting the value of `defaultAssetsPrefix`. If `defaultAssetsPrefix` is not set, when the file extension is not found in the `assetsPrefix` object, it will be set to the empty string `''` by default.

Reviewer: ```suggestion
The object key is the file extension, and its value is the CDN prefix path. You can set the default values of all other files by setting the value of `defaultAssetsPrefix`. If `defaultAssetsPrefix` is not set, when the file extension is not found in the `assetsPrefix` object, it will be set to the empty string `''` by default.
```
A little bit of rewording
